doc/src/manual/interfaces.md

```diff
@@ -371,3 +371,190 @@ If you are defining an array type that allows non-traditional indexing (indices
 something other than 1), you should specialize `indices`. You should also specialize [`similar`](@ref)
 so that the `dims` argument (ordinarily a `Dims` size-tuple) can accept `AbstractUnitRange` objects,
 perhaps range-types `Ind` of your own design. For more information, see [Arrays with custom indices](@ref).
+
+## Specializing broadcasting
+
+| Methods to implement | Brief description |
+|:-------------------- |:----------------- |
+| `Broadcast.rule(::Type{SrcType}) = ContainerType` | Output type produced by broadcasting `SrcType` |
```

Comment:
Yes to here, maybe no to below. Should I call this `ContainerStyle` or `ContainerSignalingType` or something else that indicates that it doesn't have to have any kind of direct relationship with the actual created container type?

Reply:
Hmm... Given that in most cases it will actually be the type of the container, `ContainerType` sounds a good choice to me. Though the description could mention that it's passed to `similar`. Anyway the text below is quite clear.

Reply:
`ContainerStyle`, `ContainerSignal`, or in any case something more general than `ContainerType` seems preferable by nature of being more accurate. To illustrate, `ArrayConflict` and `PromoteToSparse` are confusing as `ContainerType`s, but not as `ContainerSignal`s or something similar.